### PR TITLE
Integrate character index with the backend

### LINKED ISSUE #105: Integrate character index page with backend
Add backend returned props to character index page and hook it up with UI

### PR DESCRIPTION
**Pull request by luigibesani**
Closes #105 
## Description 
The data originally displayed on the character page was provided in the form of mock data. This has been replaced and the data displayed on the character page is obtained through the props passed in the Inertia render request.

## Changes made
-Removed character mock data
-Added new type 'CharacterRowData' to handle the data acquired from the database
-Adjust the Character and CharacterTable components to use the Inertia passed props instead of mock data
-Refactor few Character variable names to simply and/or better reveal purpose

## How you can test it
- run 'php artisan migrate:fresh && php artisan db:seed'
- login with an existing user:
-- 1. run 'php artisan tinker'
-- 2. use command 'User::first()' to obtain the first user's email
-- 3. Log in using the user's email and password : password
- navigate to localhost/character to display a table of characters assigned to the logged in user

### CODE REVIEW

File: resources/js/Components/Table/CharacterTable/CharacterTable.tsx [hunk 1]
```diff
@@ -96,7 +96,7 @@ const CharacterTable = ({
 
     const handleSelectAllClick = (event: any) => {
         if (event.target.checked) {
-            const newSelecteds = rows.map((n) => n.cname)
+            const newSelecteds = rows.map((n) => n.name)
```
Comment (willyyhuang) [inline resources/js/Components/Table/CharacterTable/CharacterTable.tsx:99]: i don't see why map is needed, if select all is just selecting all rows

File: resources/js/Components/Table/CharacterTable/CharacterTable.tsx [hunk 2]
```diff
@@ -167,12 +167,12 @@ const CharacterTable = ({
                                 page * rowsPerPage,
                                 page * rowsPerPage + rowsPerPage,
                             )
-                            .map((row: RowData, index: number) => {
-                                const isItemSelected = isSelected(row.cname)
+                            .map((row: CharacterRowData, index: number) => {
+                                const isItemSelected = isSelected(row.name)
```
Comment (willyyhuang) [inline resources/js/Components/Table/CharacterTable/CharacterTable.tsx:171]: can you also refactor the isSelected function to make it use `.includes` instead

File: resources/js/Components/Table/CharacterTable/CharacterTable.tsx [hunk 3]
```diff
@@ -167,12 +167,12 @@ const CharacterTable = ({
...
                                 const labelId = `enhanced-table-checkbox-${index}`
                                 return (
                                     <TableRow
-                                        key={row.cname}
+                                        key={row.name}
```
Comment (willyyhuang) [inline resources/js/Components/Table/CharacterTable/CharacterTable.tsx:175]: i believe key should be the id from backend somehow, the name is not an unique identifier

File: resources/js/Components/Table/CharacterTable/CharacterTable.tsx (hunk 1 above)
Comment (luigibesani) [reply resources/js/Components/Table/CharacterTable/CharacterTable.tsx:99]: It seems that rows are set based on the 'selected' array of strings. 'newSelected' contains a list of names of all the characters.  Do you know of another way of strictly extracting the 'names' from the 'rows' array?
image

File: resources/js/Components/Table/CharacterTable/CharacterTable.tsx [hunk 4]
```diff
@@ -122,7 +122,7 @@ const CharacterTable = ({
         setSelected(newSelected)
     }
 
-    const isSelected = (name: string) => selected.indexOf(name) !== -1
+    const isSelected = (name: string) => selected.includes(name)
```
Comment (luigibesani) [inline resources/js/Components/Table/CharacterTable/CharacterTable.tsx:125]: Addressed the 'isSelected' requested change here @willyyhuang

File: resources/js/Components/Table/CharacterTable/CharacterTable.tsx [hunk 5]
```diff
@@ -122,7 +123,7 @@ const CharacterTable = ({
         setSelected(newSelected)
     }
 
-    const isSelected = (name: string) => selected.indexOf(name) !== -1
+    const isSelected = (id: number) => selected.includes(id)
```
Comment (willyyhuang) [inline resources/js/Components/Table/CharacterTable/CharacterTable.tsx:126]: please move this since this is not re-usable

**Review by willyyhuang — COMMENTED** [review]
one little thing left :)

**Comment by willyyhuang** [comment]
also unit test failed

File: resources/js/Components/Table/CharacterTable/CharacterTable.tsx (hunk 5 above)
Comment (luigibesani) [reply resources/js/Components/Table/CharacterTable/CharacterTable.tsx:126]: I thought I removed this in my last commit lol! Will remove it for sure this time.

**Review by willyyhuang — APPROVED** [review]

**Pull request merged**